sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParameters.scala

```diff
@@ -73,6 +82,12 @@ private[sql] case class CSVParameters(@transient parameters: Map[String, String]
 
   val nullValue = parameters.getOrElse("nullValue", "")
 
+  val compressionCodec: String = {
+    val maybeCodecName =
+      Option(parameters.getOrElse("compression", parameters.getOrElse("codec", null)))
+    maybeCodecName.map(_.toLowerCase).map(shortCompressionCodecNames).orNull
```

Comment:
this logic is becoming confusing with so many level of nesting. we should rewrite it (even with more loc) to make it more readable.